R/as-ped-cr.R

```diff
@@ -0,0 +1,195 @@
+#' different for all associated risks, e.g. the user can supply distinct cut 
+#' points for each single risk. This function returns either a data.frame 
+#' which complies with the data.frames proposed 
+#' \href{https://arxiv.org/abs/2006.15442}{here} or a list of single risk 
+#' data.frames. (This depends on the argument "output".)
+#' Currently our function only supports cause-specific competing 
+#' risks. That means that for each single risk a competing event is treated 
+#' (and recoded) as censoring event.
+#'
+#' @rdname as_ped_cr
+#' @param data Either an object inheriting from data frame or in case of
+#' time-dependent covariates a list of data frames (of length 2), where the 
+#' first data frame contains the time-to-event information and static 
+#' covariates while the second contains information on time-dependent
+#' covariates and the times at which they have been observed.
+#' @param formula A two sided formula with a \code{\link[survival]{Surv}} object
+#' on the left-hand-side and covariate specification on the right-hand-side (RHS).
+#' The RHS can be an extended formula, which specifies how TDCs should be transformed
+#' using specials \code{concurrent} and \code{cumulative}.
+#' @param cut Either a numeric vector or a list of numeric vectors.
+#' If provided as list, the length of the list must equal the number of 
+#' competing risks.
+#' Break points, used to partition the follow up into intervals.
+#' If unspecified, all unique event times will be used.
+#' @param max_time If \code{cut} is unspecified, this will be the last
+#' possible event time. All event times after \code{max_time}
+#' will be administratively censored at \code{max_time}.
+#' @param censor_code Either a string or integer (depending on data) with out
+#' outlines which level in the status variable is associated with censorship.
+#' @param output A chracter value: either "data.frame" or "list". 
+#' Will be matched via \code{match.arg}.
+#' @param ... Further arguments passed to the \code{as_ped} function or its 
+#' methods (\code{data.frame} method) and eventually to 
+#' \code{\link[survival]{survSplit}}
+#' @importFrom Formula Formula
+#' @examples
+#' library(mvna)
+#' data("sir.adm", package = "mvna")
+#' sir_adm <- sir.adm[c(1, 36, 85), ]
+#' sir_adm %>% as_ped_cr(Surv(time, status) ~ age + sex, 
+#'                       cut = list(c(0, 5, 20), c(0, 10, 25)))
+#' sir_adm %>% as_ped_cr(Surv(time, status) ~ age + sex, output = "list")
+#' sir_adm$status <- c("death", "discharge", "cens")
+#' sir_adm %>% as_ped_cr(Surv(time, status) ~ age + sex, censor_code = "cens")
+#' sir_adm %>% as_ped_cr(Surv(time, status) ~ age + sex, output = "list", 
+#'                       censor_code = "cens")
+#' @return A data frame class \code{ped} in piece-wise exponential data format
+#' with an additional column indicating which cause is observed (if output == 
+#' "data.frame".) or a named list of single risk data.frames (if output == 
```

Comment:
single risk PED objects instead of data.frames?

Reply:
Good remark.